scipy/constants/codata.py

```diff
@@ -1333,33 +1705,26 @@ def find(sub=None, disp=False):
     else:
         return result
 
-
-# Table is lacking some digits for exact values: calculate from definition
-c = value('speed of light in vacuum')
-mu0 = 4e-7 * pi
-epsilon0 = 1 / (mu0 * c * c)
-
-exact_values = {
-    'mag. constant': (mu0, 'N A^-2', 0.0),
-    'electric constant': (epsilon0, 'F m^-1', 0.0),
-    'characteristic impedance of vacuum': (sqrt(mu0 / epsilon0), 'ohm', 0.0),
-    'atomic unit of permittivity': (4 * epsilon0 * pi, 'F m^-1', 0.0),
-    'joule-kilogram relationship': (1 / (c * c), 'kg', 0.0),
-    'kilogram-joule relationship': (c * c, 'J', 0.0),
-    'hertz-inverse meter relationship': (1 / c, 'm^-1', 0.0)
-}
-
-# sanity check
-for key in exact_values:
-    val = _current_constants[key][0]
-    if abs(exact_values[key][0] - val) / val > 1e-9:
-        raise ValueError("Constants.codata: exact values too far off.")
-
-physical_constants.update(exact_values)
+exact_values = dict()
+for physical_constant in physical_constants:
+    if physical_constants[physical_constant][2] == 0.0:
+         exact_values[physical_constant] = physical_constants[physical_constant]
+physical_constants.update(exact_values)        
```

Comment:
Does this line do anything? `exact_values` were taken from `physical_constants` in the first place.

Reply:
I'm not sure. I thought it wouldn't be very clever to fix the `constant values` to the selection 
```
exact_values = {
    'mag. constant': (mu0, 'N A^-2', 0.0),
    'electric constant': (epsilon0, 'F m^-1', 0.0),
    'characteristic impedance of vacuum': (sqrt(mu0 / epsilon0), 'ohm', 0.0),
    'atomic unit of permittivity': (4 * epsilon0 * pi, 'F m^-1', 0.0),
    'joule-kilogram relationship': (1 / (c * c), 'kg', 0.0),
    'kilogram-joule relationship': (c * c, 'J', 0.0),
    'hertz-inverse meter relationship': (1 / c, 'm^-1', 0.0)
}
```

that was hard coded prior to my commit. But I can reverse that if you want.

Reply:
I was mainly commenting on the fact you're updating the dictionary with values from itself which should be equivalent to doing nothing.

However, now you mention it, is it possible that calculating the equations at double precision gives more precision than reading data from the table?

Reply:
> I was mainly commenting on the fact you're updating the dictionary with values from itself which should be equivalent to doing nothing.
Now as you point it out, that's true. 

> However, now you mention it, is it possible that calculating the equations at double precision gives more precision than reading data from the table?

I'm not sure Bout that one.

Reply:
Should be fixed.